pkg/engine/plugins.go

```diff
@@ -233,38 +234,49 @@ func computeDefaultProviderPlugins(languagePlugins, allPlugins pluginSet) map[to
 			continue
 		}
 
-		if seenVersion, has := defaultProviderVersions[tokens.Package(p.Name)]; has {
-			if seenVersion == nil {
+		if seenPlugin, has := defaultProviderPlugins[tokens.Package(p.Name)]; has {
+			if seenPlugin.Version == nil {
 				logging.V(preparePluginLog).Infof(
 					"computeDefaultProviderPlugins(): plugin %s selected for package %s (override, previous was nil)",
 					p, p.Name)
-				defaultProviderVersions[tokens.Package(p.Name)] = p.Version
+				defaultProviderPlugins[tokens.Package(p.Name)] = p
 				continue
 			}
 
 			contract.Assertf(p.Version != nil, "p.Version should not be nil if sorting is correct!")
-			if p.Version != nil && p.Version.GT(*seenVersion) {
-				logging.V(preparePluginLog).Infof(
-					"computeDefaultProviderPlugins(): plugin %s selected for package %s (override, newer than previous %s)",
-					p, p.Name, seenVersion)
-				defaultProviderVersions[tokens.Package(p.Name)] = p.Version
-				continue
+			contract.Assertf(p.Version.GTE(*seenPlugin.Version),
+				"Should not have seen an older plugin if sorting is correct!\n  %s-%s %s\n  %s-%s - %s",
+				p.Name, p.Version.String(), p.Path,
+				seenPlugin.Name, seenPlugin.Version.String(), seenPlugin.Path)
```

Comment:
this basically takes the place of the original check, but now only fires if LT is true (where before it would fire with LTE on the version check).  And, if it fires, it also at least includes more information so we can see why it's firing.